PWR/PWR_RTC/Core/Src/main.c

```diff
@@ -0,0 +1,133 @@
+#include "stm32l053xx.h"
+
+//------ DEFINES
+//--- INPUTS
+#define USR_BT_PRESS		(!(GPIOA->IDR & GPIO_IDR_ID10_Msk))
+
+//--- OUTPUTS
+#define GREEN_LED_ON		(GPIOA->ODR |= GPIO_ODR_OD7_Msk)
+#define	GREEN_LED_OFF		(GPIOA->ODR &= ~GPIO_ODR_OD7_Msk)
+#define TOGGLE_GREEN_LED	(GPIOA->ODR ^=GPIO_IDR_ID7_Msk)
+
+//--- GENERAL
+#define BUTTON_DEBOUNCE	50		//constant for button debounce
+#define RELEASED		0		//user button status define
+#define	PRESSED			1		//user button status define
+#define RTC_30_SECONDS	32000	//used for RTC config
+
+//------ VARIABLES
+uint8_t FlagEXTI 	= 0,		//this flag is set when there's an EXTI 10 interrupt
+		UsrBtStatus	= 0,		//this var indicates the user button status (RELEASED or PRESSED)
+		UsrBtCount	= 0;		//counter to debounce the button
+
+void MCU_Init(void);
+
+int main(void)
+{
+	MCU_Init();
+
+	while(1)
+	{
+		if(USR_BT_PRESS)
+		{
+			if(UsrBtCount++ >= BUTTON_DEBOUNCE)	//we increment UsrBtCount until it reaches BUTTON_DEBOUNCE value
+			{
+				UsrBtStatus = PRESSED;			//then we recognize the button is definitely pressed
+				FlagEXTI = 0;					//and clear the FlagEXTI
+			}
+			else
+			{
+				UsrBtStatus = RELEASED;			//or the button is not pressed
+			}
+
+			if(UsrBtStatus == PRESSED)			//when we recognized the user button was pressed
+			{
+				if((!USR_BT_PRESS))				//if it was then released, then the uC start taking action
+				{
+					GREEN_LED_OFF;				//turn the green led off
+					UsrBtStatus = RELEASED;		//set user button status as RELEASED
+
+					PWR->CR  |=	PWR_CR_DBP;		//enable write access to the RTC and RCC CSR registers
+					RTC->CR |= RTC_CR_WUTE;		//enables RTC - starts counting
+
+					__WFI();					//stop mode - resets the uC on wake-up (check RCC_CSR_SBF)
+				}
+			}
+		}
+		else
+		{
+			UsrBtCount = 0;
+		}
+	}
+
+	return 0;
+}
+
+void MCU_Init(void)
+{
+	/*							   *
+	 * --- GPIO INITIALIZATION --- *
+	 *							   */
+	/* -> GPIOA
+	 * PA14 -> AF mode (SWD_IO)		| Low speed	| Push pull
+	 * PA13 -> AF mode (SWD_CK)		| Low speed	| Push pull
+	 * PA10	-> Input (User Button)	| Low speed	| Pull-up
+	 */
+	RCC->IOPENR	 |=	RCC_IOPENR_GPIOAEN;		//enable GPIOA clock
+
+	GPIOA->MODER &=	~GPIO_MODER_MODE10_Msk;	//set PA10 as input
+	GPIOA->PUPDR |= GPIO_PUPDR_PUPD10_0;	//enables PA10 pull-up
+
+	//Green led, debug purpose
+	GPIOA->MODER &= ~GPIO_MODER_MODE7_1;	//set PA7 as output
+	GPIOA->BSRR |= GPIO_BSRR_BR_7;			//resets the corresponding OD7 bit
+	GREEN_LED_ON;
+
+	/*									   *
+	 * --- STANDBY MODE INITIALIZATION --- *
+	 *									   */
+	RCC->APB1ENR |= RCC_APB1ENR_PWREN;	//enable PWR clock
+	SCB->SCR = SCB_SCR_SLEEPDEEP_Msk; 	//low-power mode = stop mode

```

Comment:
Is it entering standby or stop mode? I'm confused, because in one place you say standby and in the other place you say stop.

Reply:
This one is entering in standby mode. I'm sorry, I just forgot to fix the comments